packages/apps/patterns/react-composer/src/components/Markdown/Markdown.tsx

```diff
@@ -5,35 +5,47 @@
 import { markdown, markdownLanguage } from '@codemirror/lang-markdown';
 import { languages } from '@codemirror/language-data';
 import CodeMirror from '@uiw/react-codemirror';
-import React from 'react';
+import React, { useMemo } from 'react';
 import { yCollab } from 'y-codemirror.next';
 
-import { Text } from '@dxos/client';
+import { Space, Text } from '@dxos/client';
+
+import { cursorColor, SpaceProvider } from '../../yjs';
 
 export type MarkdownComposerSlots = {};
 
 export type MarkdownComposerProps = {
   text?: Text;
+  space?: Space;
   slots?: MarkdownComposerSlots;
 };
 
-export const MarkdownComposer = ({ text }: MarkdownComposerProps) => {
+export const MarkdownComposer = ({ text, space }: MarkdownComposerProps) => {
   const ytext = text?.doc?.getText('md');
 
+  const { awareness } = useMemo(() => {
+    if (!space || !text?.doc) {
+      return { awareness: null };
+    }
+
+    const provider = new SpaceProvider({ space, doc: text.doc });
+    provider.awareness.setLocalStateField('user', {
+      name: 'Anonymous ' + Math.floor(Math.random() * 100),
```

Comment:
get from client.halo?

Reply:
these are just placeholders so that it works, @thure 's gonna clean this all up including the colours :slightly_smiling_face: